File: dehaze-python/app/models/schema/member.py
from pydantic import BaseModel, Field


class MemberLevelAdjustForm(BaseModel):
    levelCode: str = Field(..., min_length=1, description="目标等级")
    expireTime: str | None = None
    reason: str = Field(default="", description="调整原因")


class MemberGrowthAdjustForm(BaseModel):
    changeValue: int = Field(..., description="变动值(正数增加/负数扣减)")
    reason: str = Field(default="", description="调整原因")


class MemberStatusForm(BaseModel):
    status: int = Field(..., ge=0, le=1, description="状态(1:正常;0:冻结)")
    reason: str | None = None


class BenefitForm(BaseModel):
    levelName: str | None = None
    growthMin: int | None = Field(default=None, ge=0)
    growthMax: int | None = Field(default=None, ge=0)
    monthlyDehazeQuota: int | None = Field(default=None, ge=0)
    monthlyEvaluateQuota: int | None = Field(default=None, ge=0)
    historyRetention: int | None = Field(default=None, ge=0)
    batchLimit: int | None = Field(default=None, ge=0)
    priority: int | None = Field(default=None, ge=1, le=4)
    advancedParams: int | None = Field(default=None, ge=0, le=1)
    hdExport: int | None = Field(default=None, ge=0, le=1)
    reportExport: int | None = Field(default=None, ge=0, le=1)
    batchDownload: int | None = Field(default=None, ge=0, le=1)
    sort: int | None = Field(default=None, ge=0)
    status: int | None = Field(default=None, ge=0, le=1)

File: dehaze-go/internal/model/bo/member.go
package bo

type MemberLevelAdjustForm struct {
	LevelCode  string  `json:"levelCode"`
	ExpireTime *string `json:"expireTime"`
	Reason     string  `json:"reason"`
}

type MemberGrowthAdjustForm struct {
	ChangeValue int `json:"changeValue"`
	// 调整原因上限 256，与 DB varchar(256) 及 Python 端 max_length 对齐，防超长触发 500
	Reason string `json:"reason" binding:"max=256"`
}

type MemberStatusForm struct {
	Status int    `json:"status"`
	Reason string `json:"reason"`
}

type BenefitForm struct {
	LevelName            *string `json:"levelName"`
	GrowthMin            *int64  `json:"growthMin" binding:"omitempty,min=0"`
	GrowthMax            *int64  `json:"growthMax" binding:"omitempty,min=0"`
	MonthlyDehazeQuota   *int    `json:"monthlyDehazeQuota" binding:"omitempty,min=0"`
	MonthlyEvaluateQuota *int    `json:"monthlyEvaluateQuota" binding:"omitempty,min=0"`
	HistoryRetention     *int    `json:"historyRetention" binding:"omitempty,min=0"`
	BatchLimit           *int    `json:"batchLimit" binding:"omitempty,min=0"`
	MaxDevices           *int    `json:"maxDevices" binding:"omitempty,min=1"`
	Priority             *int    `json:"priority" binding:"omitempty,min=1,max=4"`
	AdvancedParams       *int    `json:"advancedParams" binding:"omitempty,oneof=0 1"`
	HdExport             *int    `json:"hdExport" binding:"omitempty,oneof=0 1"`
	ReportExport         *int    `json:"reportExport" binding:"omitempty,oneof=0 1"`
	BatchDownload        *int    `json:"batchDownload" binding:"omitempty,oneof=0 1"`
	Sort                 *int    `json:"sort" binding:"omitempty,min=0"`
	Status               *int    `json:"status" binding:"omitempty,oneof=0 1"`
}
